plugins/modules/service_config.py

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/python
+# Copyright: (c) 2024, Dell Technologies
+
+# Apache License version 2.0 (see MODULE-LICENSE or http://www.apache.org/licenses/LICENSE-2.0.txt)
+
+"""Ansible module for managing Service Config on PowerStore"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: service_config
+version_added: '3.1.0'
+
+short_description: Manage Service config on PowerStore storage systems
+
+description:
+- Manage Service config on PowerStore storage systems includes
+  retrieving, and updating Service config.
+
+extends_documentation_fragment:
+  - dellemc.powerstore.powerstore
+
+author:
+- Bhavneet Sharma(@Bhavneet-Sharma) <ansible.team@dell.com>
+
+options:
+  service_config:
+    description:
+    - Specifies the appliance details on which ssh will be enabled/disabled.
+    type: list
+    elements: dict
+    suboptions:
+      appliance_name:
+        description:
+        - Specifies the name of the appliance.
+        - Mutually exclusive with appliance_id.
+        type: str
+      appliance_id:
+        description:
+        - Specifies the appliance id.
+        - Mutually exclusive with appliance_name.
+        type: str
+      is_ssh_enabled:
+        description:
+        - Whether ssh will be enabled/disabled on specified appliance.
+        type: bool
+        required: true
```

Comment:
this param will not be needed for get operation, hence setting it as required is not ideal.

Reply:
This param is required only when service_config list of dict param passed. So it required for service_config param, and not required in Get operation.